### DISCUSSION TITLE
In some situation, I get this message "sql: database is closed" and pocketbase quits

### DISCUSSION THREAD

**Discussion by square-foot** [pocketbase/pocketbase#3901] [Q&A] [answered]
I have been doing some live changes on the GoJa code on the server (Ubuntu) It does live reloading but once a while I noticed it gives a message "sql: database is closed" and quits. Anything I may be doing wrong? Thanks in advance 

<img width="238" alt="image" src="https://github.com/pocketbase/pocketbase/assets/102347087/d7907acc-575b-4234-a3db-99be217e4c09">

**Comment by ganigeorgiev** [comment, marked as answer]
It is difficult to pinpoint the exact cause without details what the code is doing but I suspect that it is because of the concurrent log request writes.

When the app restarts or terminate we start a cleanup procedure (aka. `app.ResetBootstrapState()`) and attempt to close the previously opened database connections which sometimes could cause db errors like the above in case there is a log request write happening in the background during that time.

With the next release and the refactored logs implementation it shouldn't happen anymore.

If you see the above error during regular app operation and not during restarts or termination then the issue is caused by something else but I don't know how to reproduce it.

**Reply by square-foot** [reply]
Thanks. Yes, I also think it is due to very fast saves I was making. It does not seem to happen when my GoJa code is stable.
